### PR TITLE
Update image classification

### PR DESCRIPTION
**Pull request by shippingwang**
Update Image Classification Docs and Models.
1. Update named & no named models, Put the named models files into the models_name folder.
2. Update train.py & reader.py w/ py.reader.
3. Move the learning rate file to the utils/ folder
4. Add run.sh script to start training
5. Upload pre-trained model.
6. TODO: support a hyperparameters table for every impl models info to users.

### CODE REVIEW

File: fluid/PaddleCV/image_classification/models/mobilenetv2.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            for i in range(n):
+                s=2 if s==2 and i==0 else 1
+                print("s++++++++++++++++++",s)
+                input = self.linear_bottleneck(
+                                         input=input,
+                                         output_channels = c,
+                                         expansion = t,
+                                         stride =s)
+            print("====================")
+        #conv2d 1x1 output_channels=1280 stride =1
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/models/mobilenetv2.py]: remove the un-used code, in line 56, and print()..
其他地方同。

File: fluid/PaddleCV/image_classification/models/mobilenetv2.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        #con2d 1x1 Q:FC or conv2d? dropout?
+        input = fluid.layers.dropout(x=pool,dropout_prob=0.2)
+        out = fluid.layers.fc(input,size=class_dim,act='softmax')
+        """
+        out = fluid.layers.conv2d(
+            input,
+            filter_size=1,num_filters
+            num_filters=class_dim
+        )
+        """
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/models/mobilenetv2.py]: 如果注释了请去掉。

File: fluid/PaddleCV/image_classification/models/mobilenetv2.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    # batch norm conv NCHW
+    def conv_bn_layer(self,
+                      input,
+                      num_filters,
+                      filter_size,
+                      stride=1,
+                      num_groups=1,
+                      padding=0,
+                      act=None):
+        print("=========",input.shape[3],"X",input.shape[1],num_filters,stride,num_groups,padding)
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/models/mobilenetv2.py]: 去掉这行

File: fluid/PaddleCV/image_classification/reader.py [hunk 4]
```diff
@@ -167,16 +167,17 @@ def reader():
...
+    file_list = os.path.join(data_dir, 'train.txt')
     return _reader_creator(
-        file_list, 'train', shuffle=True, color_jitter=False, rotate=False, data_dir=data_dir)
+        file_list, 'train', shuffle=True, color_jitter=False, rotate=False, data_dir=os.path.join(data_dir, 'train'))
 
 
 def val(data_dir=DATA_DIR):
-    file_list = os.path.join(data_dir, 'val_list.txt')
-    return _reader_creator(file_list, 'val', shuffle=False, data_dir=data_dir)
+    file_list = os.path.join(data_dir, 'val.txt')
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/reader.py]: 这里的名字请保持和 https://github.com/PaddlePaddle/models/blob/develop/fluid/PaddleCV/image_classification/data/ILSVRC2012/download_imagenet2012.sh#L37 

这里下载的一致

Comment (shippingwang) [reply fluid/PaddleCV/image_classification/reader.py]: 好的

> 这里的名字请保持和 https://github.com/PaddlePaddle/models/blob/develop/fluid/PaddleCV/image_classification/data/ILSVRC2012/download_imagenet2012.sh#L37
> 
> 这里下载的一致

好的

File: fluid/PaddleCV/image_classification/README.md [hunk 5]
```diff
@@ -80,6 +80,9 @@ python train.py \
 * **lr**: initialized learning rate. Default: 0.1.
 * **pretrained_model**: model path for pretraining. Default: None.
 * **checkpoint**: the checkpoint path to resume. Default: None.
+* **model_category**: the category of models, ("models"|"models_name"). Default: "models".
+
+Or can start the training step by running the ```run.sh```.
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/README.md:85]: train.sh ?

File: fluid/PaddleCV/image_classification/README.md [hunk 6]
```diff
@@ -183,26 +186,23 @@ Test-12-score: [15.040644], class [386]
 
 ## Supported models and performances
 
+Models consists of two categories: Pre-trained models with specified parameters names in model definition and Pre-trained models without specified parameters, Generate named model by indicating ```model_category = models_name```.
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/README.md]: Pre-trained models ->  models

File: fluid/PaddleCV/image_classification/models/se_resnext.py [hunk 7]
```diff
@@ -105,8 +104,7 @@ def net(self, input, class_dim=1000):
 
         pool = fluid.layers.pool2d(
             input=conv, pool_size=7, pool_type='avg', global_pooling=True)
-        drop = fluid.layers.dropout(
-            x=pool, dropout_prob=0.5, seed=self.params['dropout_seed'])
+        drop = fluid.layers.dropout(x=pool, dropout_prob=0.5)
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/models/se_resnext.py]: do not remove the `seed=self.params['dropout_seed']`

File: fluid/PaddleCV/image_classification/models/se_resnext.py [hunk 8]
```diff
@@ -14,7 +14,6 @@
     "input_size": [3, 224, 224],
     "input_mean": [0.485, 0.456, 0.406],
     "input_std": [0.229, 0.224, 0.225],
-    "dropout_seed": None,
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/models/se_resnext.py]: dot not remove the `dropout_seed`

File: fluid/PaddleCV/image_classification/README.md (hunk 5 above)
Comment (shippingwang) [reply fluid/PaddleCV/image_classification/README.md:85]: > train.sh ?

改成run.sh了

File: fluid/PaddleCV/image_classification/README.md (hunk 6 above)
Comment (shippingwang) [reply fluid/PaddleCV/image_classification/README.md]: > Pre-trained models -> models
嗯

**Review by qingqing01 — APPROVED** [review]
